scripts/seurat_alignment_qc.Rmd

```diff
@@ -213,28 +210,42 @@ head(colData(sce))
 ```{r miQC-plot, echo=TRUE}
 print(plotMetrics(sce))
 
-model <- mixtureModel(sce)
-parameters(model)
-head(posterior(model))
+run_qc <- function(sce) {
+  
+  if (any(sce$subsets_mito_percent == 0)) {
+    print("miQC did not work, use default mtDNA.pct for filtering.")
+    mtDNA.pct_default = 5
+    filtered_sce <- subset(seurat_obj, subset = pct_mito < mtDNA.pct_default)
+    
+    } else {
+      print("miQC worked.")
+      model <- mixtureModel(sce)
+      print(is.nan(model@logLik))                                    # Apply is.nan function
+      parameters(model)
+      head(posterior(model))
+      
+      # Plot
+      # The cells at the very top of the graph are almost certainly compromised, 
+      # most likely to have been derived from the distribution with fewer unique genes and higher baseline mitochondrial expression.
+      print(plotModel(sce, model))
+
+      # Plot
+      print(plotFiltering(sce, model))
+
+      # Filter and remove the indicated cells by miQC
+      filtered_sce <- filterCells(sce, model)
+    }
+  
+  }
+  
 
-# Plot
-# The cells at the very top of the graph are almost certainly compromised, most likely to have been derived from the distribution with fewer unique genes and higher baseline mitochondrial expression.
-print(plotModel(sce, model))
+# Run QC function
+p <- run_qc(sce) 
 
-# Attention: To add code for cases when miQC won't work
-# https://github.com/AlexsLemonade/scpca-nf/blob/main/bin/filter_sce.R
 ```
-                                          
 ## Filter data by min_genes
 
 ```{r data-filter, echo=TRUE}
-# Plot
-print(plotFiltering(sce, model))
-
-# Filter and remove the indicated cells by miQC
-filtered_sce <- filterCells(sce, model)
-
-
 # Convert sce back to seurat object 
 seurat_obj <- as.Seurat(filtered_sce, 
```

Comment:
Here I get error: `Error: object 'filtered_sce' not found`
Likely because what you want is in a function, but it has no `return` value.
However, in local testing, if I return `filtered_sce`, and try t o convert `p`, I get:
```
Error in UseMethod(generic = "as.Seurat", object = x) :
  no applicable method for 'as.Seurat' applied to an object of class "Seurat"
```
My guess is that `filtered_sce <- subset(seurat_obj, subset = pct_mito < mtDNA.pct_default)` is already a seurat object in the function, but `filtered_sce <- filterCells(sce, model)` might not be. 

All of that to say I think I was able to fix it by giving it a return value and also just converting within the function to seurat object in the `else` statement of the function. I will push my fix and you can review

Reply:
Thanks for catching this! Yes, that is correct, I should have changed it to `p`, which is the return value of the function.